File: LastOcc/Find.ts
/*
    Author : Pakshal Shashikant Jain 
    Date : 11/07/2023
    Program : Accept N numbers from user and accept one another number as NO ,
              return index of last occurrence of that NO.

              Input : N : 6
                      NO: 66
                      Elements : 85 66 3 66 93 88
              Output : 3

              Input : N : 6
                      NO: 93
                      Elements : 85 66 3 66 93 88
              Output : 4
            
              Input : N : 6
                      NO: 12
                      Elements : 85 11 3 15 11 111
              Output : -1 
*/

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Class Declaration
class Occurance
{
    //Characteristics of Class
    NO : number;
    arr : number[];
    isize : number;
    
    //Constructor of Class
    construct()
    {
        this.NO = 0;
        this.arr = [];
        this.isize = 0;
    }

    //Behaviour of Class to initialize value to Class Variables
    Accept()
    {
        this.isize = 6;
        this.arr = [85,66,12,12,23,88];
        this.NO = 65;
    }

    //Behaviour of Class To find Last Occurance of Entered number
    Find()
    {
        var i : number = 0;

        //Logic To find Last Occurance
        for(i = this.isize-1;i > 0;i--)
        {
            if(this.arr[i] == this.NO)
            {
                break;
            }
        }

        //Handeling Conditions
        if(this.arr[i] == this.NO)
        {
            return i;
        }
        else
        {
            return -1;
        }
    }
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
function main() 
{
    var iret : number = 0;

    console.log("Ganpati Bappa ")

    //Creating object of Occurance Class
    var obj = new Occurance();

    //Call To Behaviour of Class
    obj.Accept();
    iret = obj.Find();

    //Priting Last Occurance of Entered Number Found...
    console.log("Last Occurance of %d number is at index number %d",obj.NO,iret);
}   

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Call To Main Function
main();

File: FindLargestNumberFromBothDiagonalsOfMatrix/Find.c
/*
    Author : Pakshal Shashikant Jain
    Date : 04/08/2024
    Program : Write a program which accept matrix and return largest number from both the
              diagonals 
*/

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required libraries
#include<stdio.h>
#include<stdlib.h>

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function to find largest number from both diagonal of matrix
int MaxDiagonal(int **arr,int irow,int icol) 
{
    int i = 0;
    int j = 0;
    int imax = 0;

    int ino = 0;
    
    //Logic to find largest number from matrix 
    for(i = 0;i < irow;i++)
    {
        for(j = 0;j < icol;j++)
        {
            if((i == j)||((i + j) == (icol - 1)))
            {
                if(arr[i][j] > imax)
                {
                    imax = arr[i][j];
                }
            }
        }
    }

    //Returning largest number
    return imax;
}

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
int main() 
{
    int i = 0;
    int j = 0;
    int row = 0;
    int col = 0;
    int iret = 0;

    int **arr = NULL;    // int arr[?][?];   eg. int arr[3][4] 
    //arr is 2 dimensional array which contains 3 one dimensional array and each
    //one dimensional array contain 4 elements and each element is of type integer 

    printf("Jay Ganesh....\n");

    //Taking input from user
    printf("Enter Number of Rows : ");
    scanf("%d",&row);

    printf("Enter Number of Columns : ");
    scanf("%d",&col);

    //allocating memory for matrix
    arr = (int **)malloc(row * sizeof(int *));

    for(i = 0;i < row;i++)
    {
        arr[i] = (int *)malloc(col * sizeof(int));
    }

    //Taking elements of matarix as input from user
    printf("Enter Elements of Matrix : \n");
    for(i = 0;i < row;i++)
    {
        printf("---Enter Elements into %d Row---\n",(i + 1));
        for(j = 0;j < col;j++)
        {
            printf("Enter Element No %d : ",(j + 1));
            scanf("%d",&arr[i][j]);
        }
    }

    //Printing Elements of Matrix
    printf("Elements of Matrix are : \n");
    for(i = 0;i < row;i++)
    {
        for(j = 0;j < col;j++)
        {
            printf("%d \t",arr[i][j]);
        }
        printf("\n");
    }

    //Call to MaxDiagonal function
    iret = MaxDiagonal(arr,row,col);

    //Printing largest number
    printf("Largest number from both the diagonal is : %d",iret);

    return 0;
}